### PR TITLE
Fix onSuccess exception handling in FileTools.RequestFile

### PR DESCRIPTION
**Pull request by mrlika**
**Background**

If put under `.gltf` extension a non-json file and try to load it this `await` never resolves and leaves `scene` in loading state forever:

```typescript
try {
  await SceneLoader.AppendAsync('https://some.site.com/models/', 'bad_non_json_file.gltf', scene);
  console.log('Model loading success');
} catch (e) {
  console.log('Model loading failed');
}
```
None of `console.log` is never called in such case.

It is because `onSuccess` callback throws exception `SyntaxError: JSON.parse: unexpected character ...` here: https://github.com/BabylonJS/Babylon.js/blob/6626433abbee8f1d1e82499915645cc54b785a50/loaders/src/glTF/glTFFileLoader.ts#L546

**Solution**

The solution is to avoid pitfalls like this one globally by fixing `FileTools.RequestFile` to behave more like a promise and call `onError` if `onSuccess` fails.

### CODE REVIEW

**Comment by sebavan** [comment]
I like it, let s see what @bghgary thinks about it ?

**Comment by mrlika** [comment]
Probably it is better to re-throw error if error handler is missing:

```javascript
if (onError) {
    onError(e);
} else {
    throw e;
}
```

**Comment by bghgary** [comment]
This is like here: https://github.com/BabylonJS/Babylon.js/blob/43a9bb796520175c0460326f71aeb3cb75fa13d2/src/Loading/sceneLoader.ts#L725-L730

Ideally, we should fix this for all the callbacks (`onProgress`, `onOpened`) and not just `onSuccess`, in case any of these throw an exception. `onError` can mimic this:
https://github.com/BabylonJS/Babylon.js/blob/43a9bb796520175c0460326f71aeb3cb75fa13d2/src/Loading/sceneLoader.ts#L702-L707

**Comment by mrlika** [comment]
@bghgary, you are right. I added `onOpened` error handling because exception in it also breaks promise flow. Please check.

Error handling of `onProgress` can be skipped since it doesn't hang promise.

In overall type of error handler `onError?: (error: RequestFileError) => void` looks incorrect to me because `RequestFile` method can throw not only `RequestFileError`s.

Because of this I don't know a good way to add error handling to offline branch of this method - there is no `WebRequest` object required by `RequestFileError` constructor.

Looks like it is a good idea to change error handler type to `onError?: (error: unknown) => void` that touches about 5 files in the project and can be done in separate PR.

File: src/Misc/fileTools.ts [hunk 1]
```diff
@@ -363,11 +364,25 @@ export class FileTools {
                 }
             };
 
+            const handleError = (error: unknown) => {
+                const message = error instanceof Error ? error.message : 'Unknown error';
```
Comment (bghgary) [inline src/Misc/fileTools.ts:368]:
```suggestion
                const message = error.message || "Unknown error";
```

**Review by bghgary — COMMENTED** [review]
One minor comment. Looks good otherwise.

**Comment by bghgary** [comment]
> Error handling of onProgress can be skipped since it doesn't hang promise.

I suppose that is true. To be consistent, we shouldn't do it in the [SceneLoader](https://github.com/BabylonJS/Babylon.js/blob/43a9bb796520175c0460326f71aeb3cb75fa13d2/src/Loading/sceneLoader.ts#L883-L890) either.

> In overall type of error handler onError?: (error: RequestFileError) => void looks incorrect to me because RequestFile method can throw not only RequestFileErrors.
> 
> Because of this I don't know a good way to add error handling to offline branch of this method - there is no WebRequest object required by RequestFileError constructor.
> 
> Looks like it is a good idea to change error handler type to onError?: (error: unknown) => void that touches about 5 files in the project and can be done in separate PR.

It's a bit more complicated than it seems as this will propagate all the way to the scene loader plugins. I'm not sure updating this is worth it.

**Review by bghgary — APPROVED** [review]

**Comment by deltakosh** [comment]
Tests are failing :(

**Comment by bghgary** [comment]
I reran them to make sure it's not from yesterday's bug.

**Comment by mrlika** [comment]
Merged with master again. Test pass.

**Review by bghgary — APPROVED** [review]

**Pull request merged**